velox/vector/arrow/Bridge.cpp

```diff
@@ -212,6 +213,33 @@ static void releaseArrowSchema(ArrowSchema* arrowSchema) {
   arrowSchema->private_data = nullptr;
 }
 
+const char* exportArrowFormatTimestampStr(
+    const ArrowOptions& options,
+    std::string& formatBuffer) {
+  switch (options.timestampUnit) {
+    case TimestampUnit::kSecond:
+      formatBuffer = "tss:";
+      break;
+    case TimestampUnit::kMilli:
+      formatBuffer = "tsm:";
+      break;
+    case TimestampUnit::kMicro:
+      formatBuffer = "tsu:";
+      break;
+    case TimestampUnit::kNano:
+      formatBuffer = "tsn:";
+      break;
+    default:
+      VELOX_UNREACHABLE();
+  }
+
+  if (options.timestampTimeZone.has_value()) {
```

Comment:
This should not happen. VELOX_DCHECK(options.timestampTimeZone.has_value())

Reply:
https://github.com/facebookincubator/velox/pull/10474#discussion_r1697340467 Update the default value to std::nullopt. And then we need this check.